--- work/sources/starlette/routing/__init__.py
+++ work/patched/starlette/routing/__init__.py
@@ -0,0 +1,26 @@
+from alpha93.starlette._internal.routing import (
+    BaseRoute,
+    Match,
+    NoMatchFound,
+    Route,
+    Router,
+    _DefaultLifespan,
+)
+
+__all__ = (
+    "BaseRoute",
+    "Match",
+    "NoMatchFound",
+    "Route",
+    "Router",
+    "_DefaultLifespan",
+    "host",
+    "mount"
+)
+
+import sys
+
+from alpha93.commons.utils import lazy
+
+sys.modules["starlette.routing.host"] = lazy("alpha93.starlette._internal.routing._routes.host")
+sys.modules["starlette.routing.mount"] = lazy("alpha93.starlette._internal.routing._routes.mount")
