--- work/sources/starlette/middleware/authentication.py	2026-05-16 04:16:01.715524108 +0000
+++ work/patched/starlette/middleware/authentication.py	2026-05-17 11:23:54.421087265 +0000
@@ -1,5 +1,3 @@
-from __future__ import annotations
-
 from collections.abc import Callable
 
 from starlette.authentication import (
@@ -10,7 +8,9 @@
 )
 from starlette.requests import HTTPConnection
 from starlette.responses import PlainTextResponse, Response
-from starlette.types import ASGIApp, Receive, Scope, Send
+
+if __debug__ and __import__("typing").TYPE_CHECKING:
+    from starlette.types import ASGIApp, Receive, Scope, Send
 
 
 class AuthenticationMiddleware:
