--- work/sources/starlette/middleware/cors.py	2026-05-16 04:16:01.715524108 +0000
+++ work/patched/starlette/middleware/cors.py	2026-05-17 11:20:21.141108279 +0000
@@ -1,12 +1,14 @@
-from __future__ import annotations
-
 import functools
 import re
-from collections.abc import Sequence
 
 from starlette.datastructures import Headers, MutableHeaders
-from starlette.responses import PlainTextResponse, Response
-from starlette.types import ASGIApp, Message, Receive, Scope, Send
+from starlette.responses import PlainTextResponse
+
+if __debug__ and __import__("typing").TYPE_CHECKING:
+    from collections.abc import Sequence
+
+    from starlette.responses import Response
+    from starlette.types import ASGIApp, Message, Receive, Scope, Send
 
 ALL_METHODS = ("DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT")
 SAFELISTED_HEADERS = {"Accept", "Accept-Language", "Content-Language", "Content-Type"}
