--- work/sources/starlette/middleware/httpsredirect.py	2026-05-16 04:16:01.719524108 +0000
+++ work/patched/starlette/middleware/httpsredirect.py	2026-05-17 11:20:21.145108279 +0000
@@ -1,6 +1,8 @@
 from starlette.datastructures import URL
 from starlette.responses import RedirectResponse
-from starlette.types import ASGIApp, Receive, Scope, Send
+
+if __debug__ and __import__("typing").TYPE_CHECKING:
+    from starlette.types import ASGIApp, Receive, Scope, Send
 
 
 class HTTPSRedirectMiddleware:
@@ -8,7 +10,7 @@
         self.app = app
 
     async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
-        if scope["type"] in ("http", "websocket") and scope["scheme"] in ("http", "ws"):
+        if scope["type"] == scope["scheme"] == "http":
             url = URL(scope=scope)
             redirect_scheme = {"http": "https", "ws": "wss"}[url.scheme]
             netloc = url.hostname if url.port in (80, 443) else url.netloc
