--- work/sources/fastapi/dependencies/utils.py	2026-04-30 16:45:51.818075012 +0900
+++ work/patched/fastapi/dependencies/utils.py	2026-05-02 16:54:54.044843569 +0900
@@ -488,12 +488,12 @@
     if depends is None and lenient_issubclass(
         type_annotation,
         (
-            Request,
-            WebSocket,
-            HTTPConnection,
-            Response,
-            StarletteBackgroundTasks,
-            SecurityScopes,
+                Request,
+                WebSocket,
+                HTTPConnection,
+                Response,
+                StarletteBackgroundTasks,
+                SecurityScopes,
         ),
     ):
         assert field_info is None, (
@@ -765,14 +765,14 @@
     if (
         value is None
         or (
-            isinstance(field.field_info, params.Form)
-            and isinstance(value, str)  # For type checks
-            and value == ""
-        )
+        isinstance(field.field_info, params.Form)
+        and isinstance(value, str)  # For type checks
+        and value == ""
+    )
         or (
-            field_annotation_is_sequence(field.field_info.annotation)
-            and len(value) == 0
-        )
+        field_annotation_is_sequence(field.field_info.annotation)
+        and len(value) == 0
+    )
     ):
         if field.field_info.is_required():
             return
