--- work/sources/src/fastmcp/utilities/exceptions.py	2026-05-03 12:29:52.963915856 +0900
+++ work/patched/src/fastmcp/utilities/exceptions.py	2026-05-03 13:32:59.890864484 +0900
@@ -43,7 +43,4 @@
     type[BaseException] | Iterable[type[BaseException]],
     Callable[[BaseExceptionGroup[Any]], Any],
 ]:
-    if fastmcp.settings.client_raise_first_exceptiongroup_error:
-        return _catch_handlers
-    else:
-        return {}
+    return _catch_handlers
