--- work/sources/starlette/applications.pyi	1970-01-01 00:00:00.000000000 +0000
+++ work/patched/starlette/applications.pyi	1970-01-01 00:00:00.000000000 +0000
@@ -0,0 +1,94 @@
+from collections.abc import Sequence, Callable, MutableSequence, Iterable
+from typing import Any, Self, ParamSpec, overload
+
+from commons.types import AwaitableOr
+from .datastructures import State, URLPath
+from .middleware import Middleware, _MiddlewareFactory
+from .requests import Request
+from .responses import Response
+from .routing import BaseRoute, Router
+from .types import ASGIApp, ExceptionHandler, ExceptionHandlers, Lifespan, Send, Receive, Scope
+
+class Starlette:
+    """A Starlette ASGI application."""
+
+    debug: bool
+    state: State
+    router: Router
+    exception_handlers: ExceptionHandlers
+    user_middleware: MutableSequence[Middleware]
+    middleware_stack: ASGIApp | None
+
+    def __init__(
+        self,
+        debug: bool = False,
+        routes: Sequence[BaseRoute] | None = None,
+        middleware: Sequence[Middleware] | None = None,
+        exception_handlers: ExceptionHandlers | None = None,
+        lifespan: Lifespan[Self] | None = None,
+    ) -> None:
+        """Initializes the application.
+
+        Parameters:
+            debug: Boolean indicating if debug tracebacks should be returned on errors.
+            routes: A list of routes to serve incoming HTTP and WebSocket requests.
+            middleware: A list of middleware to run for every request. A starlette
+                application will always automatically include two middleware classes.
+                `ServerErrorMiddleware` is added as the very outermost middleware, to handle
+                any uncaught errors occurring anywhere in the entire stack.
+                `ExceptionMiddleware` is added as the very innermost middleware, to deal
+                with handled exception cases occurring in the routing or endpoints.
+            exception_handlers: A mapping of either integer status codes,
+                or exception class types onto callables which handle the exceptions.
+                Exception handler callables should be of the form
+                `handler(request, exc) -> response` and may be either standard functions, or
+                async functions.
+            lifespan: A lifespan context function, which can be used to perform
+                startup and shutdown tasks. This is a newer style that replaces the
+                `on_startup` and `on_shutdown` handlers. Use one or the other, not both.
+        """
+        ...
+
+    async def __call__(self, scope: Scope, receive: Receive, send: Send, /) -> None: ...
+
+    def build_middleware_stack(self, /) -> ASGIApp: ...
+    def url_path_for(self, name: str, /, **path_params: Any) -> URLPath: ...
+    def add_middleware[**P](self, cls: _MiddlewareFactory[P], *args: P.args, **kwargs: P.kwargs) -> None: ...
+
+    def mount(self, path: str, app: ASGIApp, name: str | None = None, /) -> None: ...
+    def host(self, host: str, app: ASGIApp, name: str | None = None, /) -> None: ...
+    def add_route(
+        self,
+        path: str,
+        endpoint: Callable[[Request], AwaitableOr[Response]],
+        /,
+        *,
+        methods: Iterable[str] | None = None,
+        name: str | None = None,
+        include_in_schema: bool = True,
+    ) -> None: ...
+
+    @overload
+    def add_exception_handler(
+        self,
+        status_code: int,
+        handler: ExceptionHandler[int],
+        /
+    ) -> None: ...
+    @overload
+    def add_exception_handler[T : BaseException](
+        self,
+        exc_type: type[T],
+        handler: ExceptionHandler[T],
+        /
+    ) -> None: ...
+    @overload
+    def add_exception_handler(
+        self,
+        status_code_or_exc_type: int | type[BaseException],
+        handler: ExceptionHandler[int | BaseException],
+        /
+    ) -> None: ...
+
+    @property
+    def routes(self, /) -> list[BaseRoute]: ...
