--- work/sources/starlette/convertors.py	2026-05-16 04:16:01.715524108 +0000
+++ work/patched/starlette/convertors.py	2026-05-17 11:20:21.137108280 +0000
@@ -1,13 +1,11 @@
-from __future__ import annotations
+if __debug__ and __import__("typing").TYPE_CHECKING:
+    from typing import Any, ClassVar
 
 import math
 import uuid
-from typing import Any, ClassVar, Generic, TypeVar
 
-T = TypeVar("T")
 
-
-class Convertor(Generic[T]):
+class Convertor[T]:
     regex: ClassVar[str] = ""
 
     def convert(self, value: str) -> T:
