--- work/sources/starlette/convertors.py	2026-04-30 16:47:44.146039129 +0900
+++ work/patched/starlette/convertors.py	2026-05-01 14:17:09.947943145 +0900
@@ -1,13 +1,11 @@
-from __future__ import annotations
+if __debug__ and __import__("typing").TYPE_CHECKING:
+    from typing import Any, ClassVar
 
 import math
 import uuid
-from typing import Any, ClassVar, Generic, TypeVar
 
-T = TypeVar("T")
 
-
-class Convertor(Generic[T]):
+class Convertor[T]:
     regex: ClassVar[str] = ""
 
     def convert(self, value: str) -> T:
