--- work/sources/starlette/requests.py	2026-04-30 16:47:44.146039129 +0900
+++ work/patched/starlette/requests.py	2026-05-02 22:44:14.792164536 +0900
@@ -1,38 +1,19 @@
-from __future__ import annotations
-
-import json
-import sys
-from collections.abc import AsyncGenerator, Iterator, Mapping
+from collections.abc import Mapping
 from http import cookies as http_cookies
-from typing import TYPE_CHECKING, Any, Generic, NoReturn, cast
+from typing import Any
 
 import anyio
 
-from starlette._utils import AwaitableOrContextManager, AwaitableOrContextManagerWrapper
-from starlette.datastructures import URL, Address, FormData, Headers, QueryParams, State
-from starlette.exceptions import HTTPException
-from starlette.formparsers import FormParser, MultiPartException, MultiPartParser
-from starlette.types import Message, Receive, Scope, Send
-
-if TYPE_CHECKING:
-    from python_multipart.multipart import parse_options_header
-
-    from starlette.applications import Starlette
-    from starlette.middleware.sessions import Session
-    from starlette.routing import Router
-else:
-    try:
-        try:
-            from python_multipart.multipart import parse_options_header
-        except ModuleNotFoundError:  # pragma: no cover
-            from multipart.multipart import parse_options_header
-    except ModuleNotFoundError:  # pragma: no cover
-        parse_options_header = None
-
-if sys.version_info >= (3, 13):  # pragma: no cover
-    from typing import TypeVar
-else:  # pragma: no cover
-    from typing_extensions import TypeVar
+from .datastructures import URL, Address, FormData, Headers, QueryParams, State
+
+if __debug__ and __import__("typing").TYPE_CHECKING:
+    from collections.abc import AsyncGenerator, Iterator
+    from typing import NoReturn
+
+    from .applications import Starlette
+    from .routing import Router
+    from .types import Message, Receive, Scope, Send
+
 
 SERVER_PUSH_HEADERS_TO_COPY = {
     "accept",
@@ -74,17 +55,14 @@
     pass
 
 
-StateT = TypeVar("StateT", bound=Mapping[str, Any] | State, default=State)
-
-
-class HTTPConnection(Mapping[str, Any], Generic[StateT]):
+class HTTPConnection[StateT : State | Mapping = State](Mapping[str, Any]):
     """
     A base class for incoming HTTP connections, that is used to provide
     any functionality that is common to both `Request` and `WebSocket`.
     """
 
     def __init__(self, scope: Scope, receive: Receive | None = None) -> None:
-        assert scope["type"] in ("http", "websocket")
+        assert scope["type"] == "http"
         self.scope = scope
 
     def __getitem__(self, key: str) -> Any:
@@ -162,19 +140,11 @@
         # client is a 2 item tuple of (host, port), None if missing
         host_port = self.scope.get("client")
         if host_port is not None:
+            host_port: Any
             return Address(*host_port)
         return None
 
     @property
-    def session(self) -> dict[str, Any]:
-        assert "session" in self.scope, "SessionMiddleware must be installed to access request.session"
-        session: Session = self.scope["session"]
-        # We keep the hasattr in case people actually use their own `SessionMiddleware` implementation.
-        if hasattr(session, "mark_accessed"):  # pragma: no branch
-            session.mark_accessed()
-        return session
-
-    @property
     def auth(self) -> Any:
         assert "auth" in self.scope, "AuthenticationMiddleware must be installed to access request.auth"
         return self.scope["auth"]
@@ -192,7 +162,7 @@
             # Create a state instance with a reference to the dict in which it should
             # store info
             self._state = State(self.scope["state"])
-        return cast(StateT, self._state)
+        return self._state
 
     def url_for(self, name: str, /, **path_params: Any) -> URL:
         url_path_provider: Router | Starlette | None = self.scope.get("router") or self.scope.get("app")
@@ -210,7 +180,7 @@
     raise RuntimeError("Send channel has not been made available")
 
 
-class Request(HTTPConnection[StateT]):
+class Request[StateT : State | Mapping = State](HTTPConnection[StateT]):
     _form: FormData | None
 
     def __init__(self, scope: Scope, receive: Receive = empty_receive, send: Send = empty_send):
@@ -224,7 +194,7 @@
 
     @property
     def method(self) -> str:
-        return cast(str, self.scope["method"])
+        return self.scope["method"]
 
     @property
     def receive(self) -> Receive:
@@ -258,58 +228,6 @@
             self._body = b"".join(chunks)
         return self._body
 
-    async def json(self) -> Any:
-        if not hasattr(self, "_json"):  # pragma: no branch
-            body = await self.body()
-            self._json = json.loads(body)
-        return self._json
-
-    async def _get_form(
-        self,
-        *,
-        max_files: int | float = 1000,
-        max_fields: int | float = 1000,
-        max_part_size: int = 1024 * 1024,
-    ) -> FormData:
-        if self._form is None:  # pragma: no branch
-            assert parse_options_header is not None, (
-                "The `python-multipart` library must be installed to use form parsing."
-            )
-            content_type_header = self.headers.get("Content-Type")
-            content_type: bytes
-            content_type, _ = parse_options_header(content_type_header)
-            if content_type == b"multipart/form-data":
-                try:
-                    multipart_parser = MultiPartParser(
-                        self.headers,
-                        self.stream(),
-                        max_files=max_files,
-                        max_fields=max_fields,
-                        max_part_size=max_part_size,
-                    )
-                    self._form = await multipart_parser.parse()
-                except MultiPartException as exc:
-                    if "app" in self.scope:
-                        raise HTTPException(status_code=400, detail=exc.message)
-                    raise exc
-            elif content_type == b"application/x-www-form-urlencoded":
-                form_parser = FormParser(self.headers, self.stream())
-                self._form = await form_parser.parse()
-            else:
-                self._form = FormData()
-        return self._form
-
-    def form(
-        self,
-        *,
-        max_files: int | float = 1000,
-        max_fields: int | float = 1000,
-        max_part_size: int = 1024 * 1024,
-    ) -> AwaitableOrContextManager[FormData]:
-        return AwaitableOrContextManagerWrapper(
-            self._get_form(max_files=max_files, max_fields=max_fields, max_part_size=max_part_size)
-        )
-
     async def close(self) -> None:
         if self._form is not None:  # pragma: no branch
             await self._form.close()
