--- work/sources/starlette/routing.py	2026-05-16 04:16:01.719524108 +0000
+++ work/patched/starlette/routing.py	2026-05-17 11:20:21.149108278 +0000
@@ -1,29 +1,36 @@
-from __future__ import annotations
-
 import contextlib
 import functools
 import inspect
 import re
 import traceback
 import types
-import warnings
-from collections.abc import Awaitable, Callable, Collection, Generator, Sequence
-from contextlib import AbstractAsyncContextManager, AbstractContextManager, asynccontextmanager
+from abc import ABC, abstractmethod
+from contextlib import AbstractAsyncContextManager
 from enum import Enum
-from re import Pattern
-from typing import Any, TypeVar
 
-from starlette._exception_handler import wrap_app_handling_exceptions
-from starlette._utils import get_route_path, is_async_callable
-from starlette.concurrency import run_in_threadpool
-from starlette.convertors import CONVERTOR_TYPES, Convertor
-from starlette.datastructures import URL, Headers, URLPath
-from starlette.exceptions import HTTPException
-from starlette.middleware import Middleware
-from starlette.requests import Request
-from starlette.responses import PlainTextResponse, RedirectResponse, Response
-from starlette.types import ASGIApp, Lifespan, Receive, Scope, Send
-from starlette.websockets import WebSocket, WebSocketClose
+from alpha93.starlette._internal.methods import Methods
+
+from ._exception_handler import wrap_app_handling_exceptions
+from ._utils import get_route_path, is_async_callable
+from .concurrency import run_in_threadpool
+from .convertors import CONVERTOR_TYPES, Convertor
+from .datastructures import URL, Headers, URLPath
+from .exceptions import HTTPException
+from .middleware import Middleware
+from .requests import Request
+from .responses import PlainTextResponse, RedirectResponse
+
+if __debug__ and __import__("typing").TYPE_CHECKING:
+    from collections.abc import Awaitable, Callable, Collection, Generator, Sequence
+    from contextlib import AbstractContextManager
+    from re import Pattern
+    from typing import Any, Final, Self
+
+    from alpha93.starlette._internal.methods import LiteralMethods
+    from commons.types import AwaitableOr
+
+    from .responses import Response
+    from .types import ASGIApp, Lifespan, Receive, Scope, Send
 
 
 class NoMatchFound(Exception):
@@ -43,9 +50,7 @@
     FULL = 2
 
 
-def request_response(
-    func: Callable[[Request], Awaitable[Response] | Response],
-) -> ASGIApp:
+def request_response(func: Callable[[Request], AwaitableOr[Response]], /) -> ASGIApp:
     """
     Takes a function or coroutine `func(request) -> response`,
     and returns an ASGI application.
@@ -54,10 +59,10 @@
         func if is_async_callable(func) else functools.partial(run_in_threadpool, func)
     )
 
-    async def app(scope: Scope, receive: Receive, send: Send) -> None:
+    async def app(scope: Scope, receive: Receive, send: Send, /) -> None:
         request = Request(scope, receive, send)
 
-        async def app(scope: Scope, receive: Receive, send: Send) -> None:
+        async def app(scope: Scope, receive: Receive, send: Send, /) -> None:
             response = await f(request)
             await response(scope, receive, send)
 
@@ -66,27 +71,8 @@
     return app
 
 
-def websocket_session(
-    func: Callable[[WebSocket], Awaitable[None]],
-) -> ASGIApp:
-    """
-    Takes a coroutine `func(session)`, and returns an ASGI application.
-    """
-    # assert asyncio.iscoroutinefunction(func), "WebSocket endpoints must be async"
-
-    async def app(scope: Scope, receive: Receive, send: Send) -> None:
-        session = WebSocket(scope, receive=receive, send=send)
-
-        async def app(scope: Scope, receive: Receive, send: Send) -> None:
-            await func(session)
-
-        await wrap_app_handling_exceptions(app, session)(scope, receive, send)
-
-    return app
-
-
-def get_name(endpoint: Callable[..., Any]) -> str:
-    return getattr(endpoint, "__name__", endpoint.__class__.__name__)
+# noinspection PyTypeChecker
+get_name: Final[Callable[[Callable[..., Any]], str]] = lambda endpoint: getattr(endpoint, "__name__", type(endpoint).__name__)
 
 
 def replace_params(
@@ -107,9 +93,7 @@
 PARAM_REGEX = re.compile("{([a-zA-Z_][a-zA-Z0-9_]*)(:[a-zA-Z_][a-zA-Z0-9_]*)?}")
 
 
-def compile_path(
-    path: str,
-) -> tuple[Pattern[str], str, dict[str, Convertor[Any]]]:
+def compile_path(path: str, /) -> tuple[Pattern[str], str, dict[str, Convertor[Any]]]:
     """
     Given a path string, like: "/{username:str}",
     or a host string, like: "{subdomain}.mydomain.org", return a three-tuple
@@ -163,17 +147,17 @@
     return re.compile(path_regex), path_format, param_convertors
 
 
-class BaseRoute:
-    def matches(self, scope: Scope) -> tuple[Match, Scope]:
-        raise NotImplementedError()  # pragma: no cover
+class BaseRoute(ABC):
+    @abstractmethod
+    def matches(self, scope: Scope, /) -> tuple[Match, Scope]: ...
 
-    def url_path_for(self, name: str, /, **path_params: Any) -> URLPath:
-        raise NotImplementedError()  # pragma: no cover
+    @abstractmethod
+    def url_path_for(self, name: str, /, **path_params: Any) -> URLPath: ...
 
-    async def handle(self, scope: Scope, receive: Receive, send: Send) -> None:
-        raise NotImplementedError()  # pragma: no cover
+    @abstractmethod
+    async def handle(self, scope: Scope, receive: Receive, send: Send, /) -> None: ...
 
-    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+    async def __call__(self, scope: Scope, receive: Receive, send: Send, /) -> None:
         """
         A route may be used in isolation as a stand-alone ASGI app.
         This is a somewhat contrived case, as they'll almost always be used
@@ -181,12 +165,8 @@
         """
         match, child_scope = self.matches(scope)
         if match == Match.NONE:
-            if scope["type"] == "http":
-                response = PlainTextResponse("Not Found", status_code=404)
-                await response(scope, receive, send)
-            elif scope["type"] == "websocket":  # pragma: no branch
-                websocket_close = WebSocketClose()
-                await websocket_close(scope, receive, send)
+            response = PlainTextResponse("Not Found", status_code=404)
+            await response(scope, receive, send)
             return
 
         scope.update(child_scope)
@@ -198,17 +178,16 @@
         self,
         path: str,
         endpoint: Callable[..., Any],
+        /,
         *,
         methods: Collection[str] | None = None,
         name: str | None = None,
-        include_in_schema: bool = True,
         middleware: Sequence[Middleware] | None = None,
     ) -> None:
         assert path.startswith("/"), "Routed paths must start with '/'"
         self.path = path
         self.endpoint = endpoint
         self.name = get_name(endpoint) if name is None else name
-        self.include_in_schema = include_in_schema
 
         endpoint_handler = endpoint
         while isinstance(endpoint_handler, functools.partial):
@@ -235,7 +214,7 @@
 
         self.path_regex, self.path_format, self.param_convertors = compile_path(path)
 
-    def matches(self, scope: Scope) -> tuple[Match, Scope]:
+    def matches(self, scope: Scope, /) -> tuple[Match, Scope]:
         path_params: dict[str, Any]
         if scope["type"] == "http":
             route_path = get_route_path(scope)
@@ -264,7 +243,7 @@
         assert not remaining_params
         return URLPath(path=path, protocol="http")
 
-    async def handle(self, scope: Scope, receive: Receive, send: Send) -> None:
+    async def handle(self, scope: Scope, receive: Receive, send: Send, /) -> None:
         if self.methods and scope["method"] not in self.methods:
             headers = {"Allow": ", ".join(self.methods)}
             if "app" in scope:
@@ -275,7 +254,7 @@
         else:
             await self.app(scope, receive, send)
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: Any, /) -> bool:
         return (
             isinstance(other, Route)
             and self.path == other.path
@@ -283,79 +262,13 @@
             and self.methods == other.methods
         )
 
-    def __repr__(self) -> str:
+    def __repr__(self, /) -> str:
         class_name = self.__class__.__name__
         methods = sorted(self.methods or [])
         path, name = self.path, self.name
         return f"{class_name}(path={path!r}, name={name!r}, methods={methods!r})"
 
 
-class WebSocketRoute(BaseRoute):
-    def __init__(
-        self,
-        path: str,
-        endpoint: Callable[..., Any],
-        *,
-        name: str | None = None,
-        middleware: Sequence[Middleware] | None = None,
-    ) -> None:
-        assert path.startswith("/"), "Routed paths must start with '/'"
-        self.path = path
-        self.endpoint = endpoint
-        self.name = get_name(endpoint) if name is None else name
-
-        endpoint_handler = endpoint
-        while isinstance(endpoint_handler, functools.partial):
-            endpoint_handler = endpoint_handler.func
-        if inspect.isfunction(endpoint_handler) or inspect.ismethod(endpoint_handler):
-            # Endpoint is function or method. Treat it as `func(websocket)`.
-            self.app = websocket_session(endpoint)
-        else:
-            # Endpoint is a class. Treat it as ASGI.
-            self.app = endpoint
-
-        if middleware is not None:
-            for cls, args, kwargs in reversed(middleware):
-                self.app = cls(self.app, *args, **kwargs)
-
-        self.path_regex, self.path_format, self.param_convertors = compile_path(path)
-
-    def matches(self, scope: Scope) -> tuple[Match, Scope]:
-        path_params: dict[str, Any]
-        if scope["type"] == "websocket":
-            route_path = get_route_path(scope)
-            match = self.path_regex.match(route_path)
-            if match:
-                matched_params = match.groupdict()
-                for key, value in matched_params.items():
-                    matched_params[key] = self.param_convertors[key].convert(value)
-                path_params = dict(scope.get("path_params", {}))
-                path_params.update(matched_params)
-                child_scope = {"endpoint": self.endpoint, "path_params": path_params}
-                return Match.FULL, child_scope
-        return Match.NONE, {}
-
-    def url_path_for(self, name: str, /, **path_params: Any) -> URLPath:
-        seen_params = set(path_params.keys())
-        expected_params = set(self.param_convertors.keys())
-
-        if name != self.name or seen_params != expected_params:
-            raise NoMatchFound(name, path_params)
-
-        path, remaining_params = replace_params(self.path_format, self.param_convertors, path_params)
-        assert not remaining_params
-        return URLPath(path=path, protocol="websocket")
-
-    async def handle(self, scope: Scope, receive: Receive, send: Send) -> None:
-        await self.app(scope, receive, send)
-
-    def __eq__(self, other: Any) -> bool:
-        return isinstance(other, WebSocketRoute) and self.path == other.path and self.endpoint == other.endpoint
-
-    def __repr__(self) -> str:
-        return f"{self.__class__.__name__}(path={self.path!r}, name={self.name!r})"
-
-
 class Mount(BaseRoute):
     def __init__(
         self,
@@ -381,12 +294,12 @@
         self.path_regex, self.path_format, self.param_convertors = compile_path(self.path + "/{path:path}")
 
     @property
-    def routes(self) -> list[BaseRoute]:
+    def routes(self, /) -> list[BaseRoute]:
         return getattr(self._base_app, "routes", [])
 
-    def matches(self, scope: Scope) -> tuple[Match, Scope]:
+    def matches(self, scope: Scope, /) -> tuple[Match, Scope]:
         path_params: dict[str, Any]
-        if scope["type"] in ("http", "websocket"):  # pragma: no branch
+        if scope["type"] == "http":  # pragma: no branch
             root_path = scope.get("root_path", "")
             route_path = get_route_path(scope)
             match = self.path_regex.match(route_path)
@@ -444,13 +357,13 @@
                     pass
         raise NoMatchFound(name, path_params)
 
-    async def handle(self, scope: Scope, receive: Receive, send: Send) -> None:
+    async def handle(self, scope: Scope, receive: Receive, send: Send, /) -> None:
         await self.app(scope, receive, send)
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: Any, /) -> bool:
         return isinstance(other, Mount) and self.path == other.path and self.app == other.app
 
-    def __repr__(self) -> str:
+    def __repr__(self, /) -> str:
         class_name = self.__class__.__name__
         name = self.name or ""
         return f"{class_name}(path={self.path!r}, name={name!r}, app={self.app!r})"
@@ -465,11 +378,11 @@
         self.host_regex, self.host_format, self.param_convertors = compile_path(host)
 
     @property
-    def routes(self) -> list[BaseRoute]:
+    def routes(self, /) -> list[BaseRoute]:
         return getattr(self.app, "routes", [])
 
-    def matches(self, scope: Scope) -> tuple[Match, Scope]:
-        if scope["type"] in ("http", "websocket"):  # pragma:no branch
+    def matches(self, scope: Scope, /) -> tuple[Match, Scope]:
+        if scope["type"] == "http":  # pragma:no branch
             headers = Headers(scope=scope)
             host = headers.get("host", "").split(":")[0]
             match = self.host_regex.match(host)
@@ -506,26 +419,23 @@
                     pass
         raise NoMatchFound(name, path_params)
 
-    async def handle(self, scope: Scope, receive: Receive, send: Send) -> None:
+    async def handle(self, scope: Scope, receive: Receive, send: Send, /) -> None:
         await self.app(scope, receive, send)
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: Any, /) -> bool:
         return isinstance(other, Host) and self.host == other.host and self.app == other.app
 
-    def __repr__(self) -> str:
+    def __repr__(self, /) -> str:
         class_name = self.__class__.__name__
         name = self.name or ""
         return f"{class_name}(host={self.host!r}, name={name!r}, app={self.app!r})"
 
 
-_T = TypeVar("_T")
-
-
-class _AsyncLiftContextManager(AbstractAsyncContextManager[_T]):
-    def __init__(self, cm: AbstractContextManager[_T]):
+class _AsyncLiftContextManager[_T](AbstractAsyncContextManager[_T]):
+    def __init__(self, cm: AbstractContextManager[_T], /):
         self._cm = cm
 
-    async def __aenter__(self) -> _T:
+    async def __aenter__(self, /) -> _T:
         return self._cm.__enter__()
 
     async def __aexit__(
@@ -533,12 +443,13 @@
         exc_type: type[BaseException] | None,
         exc_value: BaseException | None,
         traceback: types.TracebackType | None,
+        /,
     ) -> bool | None:
         return self._cm.__exit__(exc_type, exc_value, traceback)
 
 
 def _wrap_gen_lifespan_context(
-    lifespan_context: Callable[[Any], Generator[Any, Any, Any]],
+    lifespan_context: Callable[[Any], Generator[Any, Any, Any]], /
 ) -> Callable[[Any], AbstractAsyncContextManager[Any]]:
     cmgr = contextlib.contextmanager(lifespan_context)
 
@@ -559,7 +470,7 @@
     async def __aexit__(self, *exc_info: object) -> None:
         pass
 
-    def __call__(self: _T, app: object) -> _T:
+    def __call__(self: Self, app: object) -> Self:
         return self
 
 
@@ -578,37 +489,14 @@
         self.routes = [] if routes is None else list(routes)
         self.redirect_slashes = redirect_slashes
         self.default = self.not_found if default is None else default
-
-        if lifespan is None:
-            self.lifespan_context: Lifespan[Any] = _DefaultLifespan(self)
-
-        elif inspect.isasyncgenfunction(lifespan):
-            warnings.warn(
-                "async generator function lifespans are deprecated, "
-                "use an @contextlib.asynccontextmanager function instead",
-                DeprecationWarning,
-            )
-            self.lifespan_context = asynccontextmanager(lifespan)
-        elif inspect.isgeneratorfunction(lifespan):
-            warnings.warn(
-                "generator function lifespans are deprecated, use an @contextlib.asynccontextmanager function instead",
-                DeprecationWarning,
-            )
-            self.lifespan_context = _wrap_gen_lifespan_context(lifespan)
-        else:
-            self.lifespan_context = lifespan
+        self.lifespan_context = lifespan or _DefaultLifespan(self)
 
         self.middleware_stack = self.app
         if middleware:
             for cls, args, kwargs in reversed(middleware):
                 self.middleware_stack = cls(self.middleware_stack, *args, **kwargs)
 
-    async def not_found(self, scope: Scope, receive: Receive, send: Send) -> None:
-        if scope["type"] == "websocket":
-            websocket_close = WebSocketClose()
-            await websocket_close(scope, receive, send)
-            return
-
+    async def not_found(self, scope: Scope, receive: Receive, send: Send, /) -> None:
         # If we're running inside a starlette application then raise an
         # exception, so that the configurable exception handler can deal with
         # returning the response. For plain ASGI apps, just return the response.
@@ -626,7 +514,7 @@
                 pass
         raise NoMatchFound(name, path_params)
 
-    async def lifespan(self, scope: Scope, receive: Receive, send: Send) -> None:
+    async def lifespan(self, scope: Scope, receive: Receive, send: Send, /) -> None:
         """
         Handle ASGI lifespan messages, which allows us to manage application
         startup and shutdown events.
@@ -653,14 +541,14 @@
         else:
             await send({"type": "lifespan.shutdown.complete"})
 
-    async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
+    async def __call__(self, scope: Scope, receive: Receive, send: Send, /) -> None:
         """
         The main entry point to the Router class.
         """
         await self.middleware_stack(scope, receive, send)
 
-    async def app(self, scope: Scope, receive: Receive, send: Send) -> None:
-        assert scope["type"] in ("http", "websocket", "lifespan")
+    async def app(self, scope: Scope, receive: Receive, send: Send, /) -> None:
+        assert scope["type"] == "http" or scope["type"] == "lifespan"
 
         if "router" not in scope:
             scope["router"] = self
@@ -684,7 +572,7 @@
                 partial_scope = child_scope
 
         if partial is not None:
-            #  Handle partial matches. These are cases where an endpoint is
+            # Handle partial matches. These are cases where an endpoint is
             # able to handle the request, but is not a preferred option.
             # We use this in particular to deal with "405 Method Not Allowed".
             scope.update(partial_scope)
@@ -709,39 +597,17 @@
 
         await self.default(scope, receive, send)
 
-    def __eq__(self, other: Any) -> bool:
+    def __eq__(self, other: Any, /) -> bool:
         return isinstance(other, Router) and self.routes == other.routes
 
-    def mount(self, path: str, app: ASGIApp, name: str | None = None) -> None:  # pragma: no cover
+    def mount(self, path: str, app: ASGIApp, name: str | None = None, /) -> None:  # pragma: no cover
         route = Mount(path, app=app, name=name)
         self.routes.append(route)
 
-    def host(self, host: str, app: ASGIApp, name: str | None = None) -> None:  # pragma: no cover
+    def host(self, host: str, app: ASGIApp, name: str | None = None, /) -> None:  # pragma: no cover
         route = Host(host, app=app, name=name)
         self.routes.append(route)
 
-    def add_route(
-        self,
-        path: str,
-        endpoint: Callable[[Request], Awaitable[Response] | Response],
-        methods: Collection[str] | None = None,
-        name: str | None = None,
-        include_in_schema: bool = True,
-    ) -> None:  # pragma: no cover
-        route = Route(
-            path,
-            endpoint=endpoint,
-            methods=methods,
-            name=name,
-            include_in_schema=include_in_schema,
-        )
-        self.routes.append(route)
-
-    def add_websocket_route(
-        self,
-        path: str,
-        endpoint: Callable[[WebSocket], Awaitable[None]],
-        name: str | None = None,
-    ) -> None:  # pragma: no cover
-        route = WebSocketRoute(path, endpoint=endpoint, name=name)
+    def add_route(self, path: str, endpoint: Callable[[Request], AwaitableOr[Response]], /, **kwargs) -> None:  # pragma: no cover
+        route = Route(path, endpoint, **kwargs)
         self.routes.append(route)
