--- work/sources/fastapi/encoders.py	2026-04-30 16:45:51.818075012 +0900
+++ work/patched/fastapi/encoders.py	2026-05-02 17:01:56.104712923 +0900
@@ -15,21 +15,14 @@
 from pathlib import Path, PurePath
 from re import Pattern
 from types import GeneratorType
-from typing import Annotated, Any
+from typing import Any
 from uuid import UUID
 
-from annotated_doc import Doc
-from fastapi.exceptions import PydanticV1NotSupportedError
-from fastapi.types import IncEx
 from pydantic import BaseModel
+from pydantic.main import IncEx
 from pydantic.networks import AnyUrl, NameEmail
 from pydantic.types import SecretBytes, SecretStr
-from pydantic_core import PydanticUndefinedType
-
-from ._compat import (
-    Url,
-    is_pydantic_v1_model_instance,
-)
+from pydantic_core import PydanticUndefinedType, Url
 
 try:
     # pydantic.color.Color is deprecated since v2.0b3, but supporting for bwd-compat
@@ -127,94 +120,15 @@
 
 
 def jsonable_encoder(
-    obj: Annotated[
-        Any,
-        Doc(
-            """
-            The input object to convert to JSON.
-            """
-        ),
-    ],
-    include: Annotated[
-        IncEx | None,
-        Doc(
-            """
-            Pydantic's `include` parameter, passed to Pydantic models to set the
-            fields to include.
-            """
-        ),
-    ] = None,
-    exclude: Annotated[
-        IncEx | None,
-        Doc(
-            """
-            Pydantic's `exclude` parameter, passed to Pydantic models to set the
-            fields to exclude.
-            """
-        ),
-    ] = None,
-    by_alias: Annotated[
-        bool,
-        Doc(
-            """
-            Pydantic's `by_alias` parameter, passed to Pydantic models to define if
-            the output should use the alias names (when provided) or the Python
-            attribute names. In an API, if you set an alias, it's probably because you
-            want to use it in the result, so you probably want to leave this set to
-            `True`.
-            """
-        ),
-    ] = True,
-    exclude_unset: Annotated[
-        bool,
-        Doc(
-            """
-            Pydantic's `exclude_unset` parameter, passed to Pydantic models to define
-            if it should exclude from the output the fields that were not explicitly
-            set (and that only had their default values).
-            """
-        ),
-    ] = False,
-    exclude_defaults: Annotated[
-        bool,
-        Doc(
-            """
-            Pydantic's `exclude_defaults` parameter, passed to Pydantic models to define
-            if it should exclude from the output the fields that had the same default
-            value, even when they were explicitly set.
-            """
-        ),
-    ] = False,
-    exclude_none: Annotated[
-        bool,
-        Doc(
-            """
-            Pydantic's `exclude_none` parameter, passed to Pydantic models to define
-            if it should exclude from the output any fields that have a `None` value.
-            """
-        ),
-    ] = False,
-    custom_encoder: Annotated[
-        dict[Any, Callable[[Any], Any]] | None,
-        Doc(
-            """
-            Pydantic's `custom_encoder` parameter, passed to Pydantic models to define
-            a custom encoder.
-            """
-        ),
-    ] = None,
-    sqlalchemy_safe: Annotated[
-        bool,
-        Doc(
-            """
-            Exclude from the output any fields that start with the name `_sa`.
-
-            This is mainly a hack for compatibility with SQLAlchemy objects, they
-            store internal SQLAlchemy-specific state in attributes named with `_sa`,
-            and those objects can't (and shouldn't be) serialized to JSON.
-            """
-        ),
-    ] = True,
+    obj,
+    include: IncEx | None = None,
+    exclude: IncEx | None = None,
+    by_alias: bool = True,
+    exclude_unset: bool = False,
+    exclude_defaults: bool = False,
+    exclude_none: bool = False,
+    custom_encoder: dict[Any, Callable[[Any], Any]] | None = None,
+    sqlalchemy_safe: bool = True,
 ) -> Any:
     """
     Convert any object to something that can be encoded in JSON.
@@ -336,11 +250,6 @@
     for encoder, classes_tuple in encoders_by_class_tuples.items():
         if isinstance(obj, classes_tuple):
             return encoder(obj)
-    if is_pydantic_v1_model_instance(obj):
-        raise PydanticV1NotSupportedError(
-            "pydantic.v1 models are no longer supported by FastAPI."
-            f" Please update the model {obj!r}."
-        )
     try:
         data = dict(obj)
     except Exception as e:
