--- work/sources/fastapi/types.pyi
+++ work/patched/fastapi/types.pyi
@@ -0,0 +1,15 @@
+from collections.abc import Callable
+from enum import Enum
+from types import UnionType
+from typing import Any
+
+from pydantic import BaseModel
+
+from .routing import APIRoute
+
+type ModelNameMap = dict[type[BaseModel] | type[Enum], str]
+type DependencyCacheKey = tuple[Callable[..., Any] | None, tuple[str, ...], str]
+
+type GenerateUniqueIdFunction = Callable[[APIRoute], str]
+
+__all__ = ("UnionType", "ModelNameMap", "DependencyCacheKey", "GenerateUniqueIdFunction", )
